### PR TITLE
feat: support mail uri

### PR DESCRIPTION
**Pull request by minh-rakuten**
# Description
Support mail uri for miniapp and external webview.

## Links
MINI-1391

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt [hunk 1]
```diff
@@ -48,4 +47,15 @@ internal class MiniAppScheme private constructor(miniAppId: String) {
         it.data = url.toUri()
         context.startActivity(it)
     }
+
+    internal fun openMaiComposer(context: Context, url: String) = Intent(Intent.ACTION_SEND).let {
+        val mail = MailTo.parse(url)
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt:52]: Can this throw an error if the string isn't a proper mail URI? Maybe it should be inside try/catch?

Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt:52]: We only call this function when it's `mailto:` uri. In addition, when there is any invalid argument, the map header of `mail` will have null value which is allowed in android mail intent.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt [hunk 2]
```diff
@@ -48,4 +47,15 @@ internal class MiniAppScheme private constructor(miniAppId: String) {
         it.data = url.toUri()
         context.startActivity(it)
     }
+
+    internal fun openMaiComposer(context: Context, url: String) = Intent(Intent.ACTION_SEND).let {
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt:51]: Looks like, you wanted to write `openMailComposer` ?

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt [hunk 3]
```diff
@@ -48,4 +47,15 @@ internal class MiniAppScheme private constructor(miniAppId: String) {
...
+        val mail = MailTo.parse(url)
+        it.putExtra(Intent.EXTRA_EMAIL, arrayOf(mail.to))
+        it.putExtra(Intent.EXTRA_TEXT, mail.body)
+        it.putExtra(Intent.EXTRA_SUBJECT, mail.subject)
+        it.putExtra(Intent.EXTRA_CC, arrayOf(mail.cc))
+        it.putExtra(Intent.EXTRA_BCC, arrayOf(mail.headers["bcc"]))
+        it.type = "message/rfc822"
+        context.startActivity(it)
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt:59]: Could we check if the `context` is okay before starting it with the intent e.g.

```
if (it.resolveActivity(context.packageManager) != null) {
   context.startActivity(it)
}
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,7 @@
 **SDK**
 - **Deprecated:** Old `getUniqueId`interface.
 - **Feature:** Added `getUniqueId` new interface for invoking data using `onSuccess` and `onError`.
+- **Feature:** Support `mailto` uri.
```
Comment (khairul-alam-licon) [inline CHANGELOG.md:7]: Maybe we can elaborate it more as following:

Added support for URI's using `mailto` e.g. `<a href="mailto:test@example.com">send email</a>` for the MiniApps.

**Review by khairul-alam-licon — COMMENTED** [review]
Looking good, I have left few minor suggestions.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt (hunk 3 above)
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppScheme.kt]: Good idea.

File: CHANGELOG.md (hunk 4 above)
Comment (minh-rakuten) [reply CHANGELOG.md]: updated with the referral link.

**Review by corycaywood — APPROVED** [review]
LGTM 👍

**Pull request merged**